frontend-v2/src/features/data/MapDosing.tsx

```diff
@@ -39,10 +37,7 @@ const MapDosing: FC<IMapDosing> = ({
 
   // Check if the uploaded CSV had rows containing non-zero amounts.
   // Use a ref to persist the initial value across renders.
-  const hasDosingRowsRef = useRef(
-    amountField !== undefined &&
-      state.data.every((row) => row[amountField] !== "."),
-  );
+  const hasDosingRowsRef = useRef(validateDosingRows(state));
```

Comment:
This will still reset when the dosing step unmounts, so it might be better if this flag was hoisted up into global state for the uploaded CSV eg. `state.hasDosingRows`.

Reply:
Otherwise, this is always true once you've completed the dosing step and added dosing rows to the data.